command/debug.go

```diff
@@ -524,140 +499,225 @@ func (c *DebugCommand) captureStaticTargets(client *api.Client) error {
-		case <-metricsIntervalTicker:
-			mIdxCount++
-			c.pollingWg.Add(1)
-			go c.metricsIntervalCapture(client, mIdxCount, errCh)
-		case <-durationCh:
-			break POLLING
-		case <-c.ShutdownCh:
-			c.UI.Info("Caught interrupt signal, exiting...")
-			interrupted = true
-			break POLLING
-		}
+	// Collect replication status if target is specified
+	if strutil.StrListContains(c.flagTargets, "replication-status") {
+		g.Add(func() error {
+			c.collectReplicationStatus(ctx)
+			return nil
+		}, func(error) {
+			cancelFunc()
+		})
 	}
 
-	// Wait for polling requests to finish if it's not an interrupt
-	if !interrupted {
-		c.pollingWg.Wait()
+	// Collect server status if target is specified
+	if strutil.StrListContains(c.flagTargets, "server-status") {
+		g.Add(func() error {
+			c.collectServerStatus(ctx)
+			return nil
+		}, func(error) {
+			cancelFunc()
+		})
 	}
 
-	// Close the done channel to signal termination, make sure collection
-	// goroutine is terminated before proceeding to persisting the info.
-	close(c.doneCh)
-	<-collectCh
+	// We shouldn't bump across errors since none is returned by the interrupts,
+	// but we error check for sanity here.
+	if err := g.Run(); err != nil {
+		return err
+	}
 
 	// Write collected data to their corresponding files
-	if err := c.persistCollection(metricsCollection, "metrics.json"); err != nil {
+	if err := c.persistCollection(c.metricsCollection, "metrics.json"); err != nil {
 		c.UI.Error(fmt.Sprintf("Error writing data to %s: %v", "metrics.json", err))
 	}
-	if err := c.persistCollection(serverStatusCollection, "server_status.json"); err != nil {
+	if err := c.persistCollection(c.serverStatusCollection, "server_status.json"); err != nil {
 		c.UI.Error(fmt.Sprintf("Error writing data to %s: %v", "server_status.json", err))
 	}
-	if err := c.persistCollection(replicationStatusCollection, "replication_status.json"); err != nil {
+	if err := c.persistCollection(c.replicationStatusCollection, "replication_status.json"); err != nil {
 		c.UI.Error(fmt.Sprintf("Error writing data to %s: %v", "replication_status.json", err))
 	}
-	if err := c.persistCollection(hostInfoCollection, "host_info.json"); err != nil {
+	if err := c.persistCollection(c.hostInfoCollection, "host_info.json"); err != nil {
 		c.UI.Error(fmt.Sprintf("Error writing data to %s: %v", "host_info.json", err))
 	}
 
 	return nil
 }
 
-func (c *DebugCommand) intervalCapture(client *api.Client, idxCount int, startTime time.Time, errCh chan<- *captureError) {
-	defer c.pollingWg.Done()
+func (c *DebugCommand) collectHostInfo(ctx context.Context) {
+	idxCount := 0
 
-	var wg sync.WaitGroup
-	currentTimestamp := time.Now().UTC()
+	// intervalTicker needs to be buffered first in order to trigger a run
+	// right away.
+	intervalTicker := make(chan time.Time, 1)
+	intervalTicker <- time.Now()
+	tick := time.NewTicker(c.flagInterval)
+	go func() {
+		for t := range tick.C {
+			intervalTicker <- t
+		}
+	}()
```

Comment:
instead of doing this you could do: 
```
for {
	// make API CALL

	select {
	case <-ctx.Done():
		tick.Stop()
		return
	case <-tick:
	}
}
```

Reply:
This would definitely be cleaner.